### PR TITLE
enhancement(OnBoardingModal): improve github url RegExp

### PR DESCRIPTION
**Pull request by Yetizone**
Resolve https://github.com/opencollective/opencollective/issues/3507

# Description

Hello, newcomer here. This PR continues the work in https://github.com/opencollective/opencollective-frontend/pull/5247
The new regex allows inputs like `xx` and `xx/qq` but not `xx/qq/zz`.
The coding style changes were automatically added when committing. 
Let me know if there's anything else that needs to be done :) 

# Screenshots
Screenshot from 2021-01-01 15-10-02
Screenshot from 2021-01-01 15-10-12
Screenshot from 2021-01-01 15-10-27
Screenshot from 2021-01-01 15-11-04

### CODE REVIEW

File: components/onboarding-modal/OnboardingModal.js [hunk 1]
```diff
@@ -241,7 +241,10 @@ class OnboardingModal extends React.Component {
     // https://github.com/shinnn/github-username-regex
     if (
       values.githubHandle !== '' &&
-      matches(values.githubHandle, /^[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}$/i) === false
+      matches(
```
Comment (SudharakaP) [inline components/onboarding-modal/OnboardingModal.js:244]: @Yetizone : Minor simplification, I think you can remove the `=== false` and negate the condition `!matches(...`

**Review by SudharakaP — CHANGES_REQUESTED** [review]
The new regex seems to have some problems I think. 🤔 

- It doesn't seem to match urls which has a `.` in them. So the example in the issue; `citizenspring/playstreets.brussels` isn't a valid string it seems. 

- GitHub allows the creation of repos with underscore `_`. But the new regex doesn't seem to allow this kinda names. So an example would be something like `playstreets_brussels`. 

- GitHub allows the creation of repos such as `playstreets--brussels` (with two `-` signs) but that doesn't seem to be supported by the regex as well. Single `-` works well.

**Comment by Yetizone** [comment]
Finished tweaking the regex to allow those repo names.

**Review by SudharakaP — APPROVED** [review]
Seems all fine to me now 👍🏽

**Review by Betree — APPROVED** [review]
Looking good, nice job @Yetizone!

**Pull request merged**